telemetry/definitions/commonDefinitions.json

```diff
@@ -1000,6 +1022,16 @@
                 { "type": "failedCount", "required": false}
             ]
         },
+        {
+            "name": "aws_filetypes_edit",
+            "description": "Use authoring features such as autocompletion, syntax checking, and highlighting, for AWS filetypes (CFN, SAM, etc.)",
+            "metadata": [
+                { "type": "result" },
+                { "type": "enabled", "required": true},
```

Comment:
If Toolkits have a way to check that authoring features are disabled, this would be `false` if user disables such features.

Reply:
Do we have at least one Toolkit that is able to disable these features today? If not, this seems like something we could add in the future once it is applicable.

Reply:
> this seems like something we could add in the future once it is applicable.

It would need to be an optional field then (in the future). In general I lean towards making fields that we know about "required" so that everything is explicit (and thus more discoverable/noticeable) in client calls.

Happy to revert this though.

Reply:
I agree with the philosophy. This field seems like YAGNI to me, but I don't have a strong preference to leave or remove it.

Reply:
I'd suggest removing it if it's not implemented. It's okay if it is made optional later on. The absence of metadata itself is a differentiating factor among metrics. Forcing callers to assign a value means we lose that additional state. In other words, metrics emitted prior to the feature being implemented would be hard to differentiate from metrics after the feature is implemented despite conveying different information.

It's interesting how many of the best-practices in writing software don't translate well for data science. We actively avoid making something optional because that's potentially one more possible state that needs to be handled. Yet optional fields encode more information that's useful for analysis. Metrics may superficially look the same, but are they semantically equivalent?

Requiring metadata essentially implies that it is known ahead of time, which is implementation-dependent. One workaround to this problem is to make the tri-state explicit: don't use a boolean, use something like `'yes' | 'no' | 'unavailable'` instead. It's pretty gross but it's one way to differentiate between the presence and absence of a functionality.

Reply:
removed